packages/ingest/consumer.js

```diff
@@ -35,9 +36,12 @@ class Consume {
 
```

Comment:
https://github.com/nasa/cumulus/pull/528/files#diff-749798afd2373f04054409a87a61b419R33
`processMessage()` doesn't wait the step function to complete, so it seems that it's still possible that more messages than configured limit are processed at the same time.

step function is started here:
https://github.com/nasa/cumulus/blob/master/packages/api/lambdas/sf-starter.js#L22

Reply:
@jennyhliu I agree however I don't think limiting concurrent executions is the intent. I think the intent is not to start them at the same time due to step function api limits (because the cumulus message adapter makes too many step function api calls). Further I _think_ there is greater scalability in the number of open executions vs the number of API calls that can be made. Some documentation to back this up:

* According to the docs, [the step function open execution limit is 1 million](https://docs.aws.amazon.com/step-functions/latest/dg/limits.html#service-limits-state-machine-executions)
* I don't really understand how AWS Step Function API does rate limiting, but it seems much more restrictive: see [Limits Related to API Action Throttling](https://docs.aws.amazon.com/step-functions/latest/dg/limits.html#service-limits-api-action-throttling)

cc @scisco 
If anyone understand the Limits Related to API Action Throttling, I would enjoy an explanation.